src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -543,6 +543,13 @@ void DiscoveryImplPlatform::HandleNodeIdResolve(void * context, MdnsService * re
     nodeData.mAddress     = result->mAddress.ValueOr({});
     nodeData.mPort        = result->mPort;
 
+    for (size_t i = 0; i < result->mTextEntrySize; ++i)
+    {
+        ByteSpan key(reinterpret_cast<const uint8_t *>(result->mTextEntries[i].mKey), strlen(result->mTextEntries[i].mKey));
```

Comment:
It appears some non-trivial effort was put in to make the mDNS code return char * strings for TextEntry keys (which should always printable be printable).  We might consider leveraging that to also use strings for node data keys.

If we don't intend to ever leverage that, we could go the other direction and slightly uncomplicate the mDNS code so it can use ByteSpans too.  It just feels very odd that the mDNS implementations are going to lengths to c-stringify keys and yet (in some cases) still be zero copy, just for the discovery code to turn these back into bytespans.

Here's the avahi implementation doing this:

```
        while (txt != nullptr)
        {
            for (size_t i = 0; i < txt->size; i++)
            {
                if (txt->text[i] == '=')
                {
                    txt->text[i] = '\0';
                    textEntries.push_back(TextEntry{ reinterpret_cast<char *>(txt->text), &txt->text[i + 1], txt->size - i - 1 });
                    break;
                }
            }
            txt = txt->next;
        }
```

Reply:
I agree it would be more natural to use `const char*` for keys - that would be more consistent with lots of DNS APIs. I didn't want to do such a refactoring as a part of this change (as the refactoring would be bigger than the change itself :)), but I can file an issue for that.

Reply:
Makes sense.  Thank you.